lib/src/http/auto_retry.dart

```diff
@@ -157,31 +157,48 @@ class RequestAutoRetry extends AutoRetryConfig {
 }
 
 /// Representation of the back-off method to use when retrying requests. A fixed
-/// back-off will space retries out by [duration]. An exponential back-off will
-/// delay retries by `d*2^n` where `d` is [duration] and `n` is the number of
+/// back-off will space retries out by [interval]. An exponential back-off will
+/// delay retries by `d*2^n` where `d` is [interval] and `n` is the number of
 /// attempts so far.
 class RetryBackOff {
   /// The base duration from which the delay between retries will be calculated.
   /// For fixed back-off, the delay will always be this value. For exponential
   /// back-off, the delay will be this value multiplied by 2^n where `n` is the
   /// number of attempts so far.
-  final Duration duration;
+  final Duration interval;
 
   /// The back-off method to use. One of none, fixed, or exponential.
   final RetryBackOffMethod method;
 
-  /// Construct a new exponential back-off representation where [duration] is
+  /// Whether to enable jitter or not.
+  final bool withJitter;
+
+  /// The maximum duration between retries.
+  final Duration maxInterval;
+
+  /// The default maximum duration between retries. (5 minutes)
+  static const Duration defaultMaxInterval = const Duration(minutes: 5);
+
+  /// Construct a new exponential back-off representation where [interval] is
   /// the base duration from which each delay will be calculated.
-  const RetryBackOff.exponential(this.duration)
-      : method = RetryBackOffMethod.exponential;
+  const RetryBackOff.exponential(this.interval,
+      {bool withJitter, Duration maxInterval})
+      : method = RetryBackOffMethod.exponential,
+        withJitter = withJitter == true,
+        maxInterval = maxInterval != null ? maxInterval : defaultMaxInterval;
```

Comment:
that is correct. It's _probably_ safe to assume most people are on a new enough version of the sdk, but I'd rather make that change all at once and at the next major release.